app/code/Transoft/Blog/Setup/InstallSchema.php

```diff
@@ -0,0 +1,97 @@
+namespace Transoft\Blog\Setup;
+
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\DB\Ddl\Table;
+use Magento\Framework\Setup\InstallSchemaInterface;
+use Magento\Framework\Setup\ModuleContextInterface;
+use Magento\Framework\Setup\SchemaSetupInterface;
+use Zend_Db_Exception;
+
+/**
+ * Class InstallSchema
+ *
+ * Setup InstallSchema
+ */
+class InstallSchema implements InstallSchemaInterface
+{
+    /**
+     * Function install
+     *
+     * @param SchemaSetupInterface $setup
+     * @param ModuleContextInterface $context
+     * @throws Zend_Db_Exception
+     */
+    public function install(
+        SchemaSetupInterface $setup,
+        ModuleContextInterface $context
+    ) {
+        $installer = $setup;
+        $installer->startSetup();
+        if (!$installer->tableExists('transoft_blog')) {
+            $table = $installer->getConnection()->newTable(
+                $installer->getTable('transoft_blog')
+            )
+                ->addColumn(
+                    'blog_id',
+                    Table::TYPE_INTEGER,
+                    null,
+                    [
+                        'identity' => true,
+                        'nullable' => false,
+                        'primary'  => true,
+                        'unsigned' => true
+                    ],
+                    'Blog ID'
+                )
+                ->addColumn(
+                    'theme',
+                    Table::TYPE_TEXT,
+                    255,
+                    ['nullable => false'],
+                    'Blog theme'
+                )
+                ->addColumn(
+                    'content',
+                    Table::TYPE_TEXT,
+                    '64k',
+                    [],
+                    'Blog Content'
+                )
+                ->addColumn(
+                    'image_url',
+                    Table::TYPE_TEXT,
+                    255,
+                    [],
+                    'Image url'
+                )
+                ->addColumn(
+                    'creation_time',
+                    Table::TYPE_TIMESTAMP,
+                    null,
+                    ['nullable' => false, 'default' => Table::TIMESTAMP_INIT],
+                    'Creation time'
+                )
+                ->setComment('Blog Table');
+            $installer->getConnection()->createTable($table);
+
+            $installer->getConnection()->addIndex(
+                $installer->getTable('transoft_blog'),
+                $setup->getIdxName(
+                    $installer->getTable('transoft_blog'),
+                    ['blog_id'],
+                    AdapterInterface::INDEX_TYPE_INDEX
+                ),
+                ['blog_id'],
+                AdapterInterface::INDEX_TYPE_INDEX
+            );
+        }
+
+        if (version_compare($context->getVersion(), '1.0.1') < 0) {
+            throw new Zend_Db_Exception('Install schema script should be upgraded');
+        }
```

Comment:
Why do we need this?

Reply:
This condition is useless, but parameter '$context' is required for 'install' method, at the same time
static test won't allow method parameters that are not used inside the actual method. So what is proper
way to fix this?

Reply:
You can suppress such static test warning (but use it only when it is really required). Add to **install** method description
@SuppressWarnings(PHPMD.UnusedFormalParameter)

Reply:
done